File: CS/DX WPF Application10/CustomSerialization.cs
using DevExpress.Utils.Serializing.Helpers;
using DevExpress.Xpf.Docking;

namespace DXWpfApplication {
    class MyDockLayoutManager : DockLayoutManager {
        protected override ISerializationController CreateSerializationController() {
            return new CustomSerializationController(this);
        }
        protected internal virtual MyCustomPanel CreateCustomPanel() {
            return new MyCustomPanel();
        }
    }
    class CustomSerializationController : SerializationController {
        public CustomSerializationController(DockLayoutManager container)
            : base(container) {
        }
        protected override BaseLayoutItem CreateItemByType(XtraPropertyInfo info, string typeStr) {
            if(typeStr == typeof(MyCustomPanel).Name)
                return ((MyDockLayoutManager)Container).CreateCustomPanel();
            return base.CreateItemByType(info, typeStr);
        }
    }
}

File: VB/DX_WPF_Application10/CustomSerialization.vb
Imports DevExpress.Utils.Serializing.Helpers
Imports DevExpress.Xpf.Docking

Namespace DXWpfApplication
	Friend Class MyDockLayoutManager
		Inherits DockLayoutManager

		Protected Overrides Function CreateSerializationController() As ISerializationController
			Return New CustomSerializationController(Me)
		End Function
		Protected Friend Overridable Function CreateCustomPanel() As MyCustomPanel
			Return New MyCustomPanel()
		End Function
	End Class
	Friend Class CustomSerializationController
		Inherits SerializationController

		Public Sub New(ByVal container As DockLayoutManager)
			MyBase.New(container)
		End Sub
		Protected Overrides Function CreateItemByType(ByVal info As XtraPropertyInfo, ByVal typeStr As String) As BaseLayoutItem
			If typeStr = GetType(MyCustomPanel).Name Then
				Return DirectCast(Container, MyDockLayoutManager).CreateCustomPanel()
			End If
			Return MyBase.CreateItemByType(info, typeStr)
		End Function
	End Class
End Namespace